### PR TITLE
docs: reorganize template docs

### PR DESCRIPTION
**Pull request by matifali**
@bpmct, it looks like #9139 was merged to [emplates-docs](https://github.com/coder/coder/tree/templates-docs). 
This PR merges it to `main`.

### CODE REVIEW

File: docs/templates/customizing.md
Comment (matifali) [inline docs/templates/customizing.md:1]: @bpmct @marcpaq, is this file needed? This looks empty.

**Comment by matifali** [comment]
Hi @matifali and @ericpaulsen, as this is a major refactor, I would like you to review if we have accidentally removed some docs. The old PR #9139 was based on a very old branch, so I had to make some decisions on what to keep.

**Comment by matifali** [comment]
@bpmct I have also checked the docs preview on coder.com and they look fine.

File: docs/templates/customizing.md
Comment (marcpaq) [reply docs/templates/customizing.md:1]: (Just for the record: This file is a parent to sub-topics, it has no content of its own except for links to its children.)

File: docs/templates/creating.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Working with templates
+
+You create and edit Coder templates as [Terraform](./concepts.md) configuration
```
Comment (marcpaq) [inline docs/templates/creating.md:3]: Should be

`You create and edit Coder templates as [Terraform](./tour.md) configuration`

File: docs/templates/tour.md [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+
+Managing the lifecycle of template resources is important. We want to make sure
+that workspaces use computing, storage, and other services efficiently.
+
+We want our workspace's home directory to persist after the workspace is stopped
+so that a developer can continue their work when they start the workspace again.
+
+We do this in 2 parts:
+
+- Our `docker_volume` resource uses the `lifecycle` block with
```
Comment (marcpaq) [inline docs/templates/tour.md:279]: Should be:

- Our `docker_volume` resource uses the `lifecycle` block with the

File: docs/templates/tour.md [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+
+```console
+> Confirm create? (yes/no) yes
+
+The template-tour template has been created! Developers can provision a workspace with this template using:
+
+   coder create --template="template-tour" [workspace name]
+```
+
+In your web browser, log in to your Coder dashboard, select **\*Templates**.
```
Comment (marcpaq) [inline docs/templates/tour.md:409]: Should be:

```
In your web browser, log in to your Coder dashboard, select **Templates**.
```

File: docs/templates/modules.md [hunk 4]
```diff
@@ -29,32 +29,32 @@ resource "coder_agent" "dev" {
...
+Learn more about
+[creating modules](https://developer.hashicorp.com/terraform/language/modules)
+and
+[module sources](https://developer.hashicorp.com/terraform/language/modules/sources)
+in the Terraform documentation.
 
 ## Git authentication
 
-If you are importing a module from a private git repository, the Coder server
-[or provisioner](../admin/provisioners.md) needs git credentials. Since this
```
Comment (marcpaq) [inline docs/templates/modules.md:41]: btw, in `docs/manifest.json`, the entry for `docs/admin/provisioners.md` should have this title:

```
          "title": "External provisioners",
```

File: docs/templates/creating.md (hunk 1 above)
Comment (matifali) [reply docs/templates/creating.md:3]: Have we removed the concepts file?

File: docs/templates/agent-metadata.md [hunk 5]
```diff
@@ -1,13 +1,12 @@
-# Agent Metadata
+# Agent metadata
 
 ![agent-metadata](../images/agent-metadata.png)
 
-With Agent Metadata, template admins can expose operational metrics from their
-workspaces to their users. It is the dynamic complement of
-[Resource Metadata](./resource-metadata.md).
```
Comment (marcpaq) [inline docs/templates/agent-metadata.md:7]: Should be:

```
[resource metadata](./resource-metadata.md).
```

File: docs/templates/resource-metadata.md [hunk 6]
```diff
@@ -91,16 +93,17 @@ resource "coder_metadata" "resource_with_icon" {
...
 - Image `/icon/image.svg`
 - Widgets `/icon/widgets.svg`
 - Database `/icon/database.svg`
 
-We also have other icons related to the IDEs. You can see all the icons
-[here](https://github.com/coder/coder/tree/main/site/static/icon).
+We also have other icons related to the IDEs. You can see more information on
+how to use the builtin icons [here](./icons.md).
 
 ## Agent Metadata
```
Comment (marcpaq) [inline docs/templates/resource-metadata.md:108]: This Agent Metadata section is redundant. I suggest deleting it since it's already covered in `docs/templates/agent-metadata.md`

File: docs/templates/open-in-coder.md [hunk 7]
```diff
@@ -13,105 +13,107 @@ Your browser does not support the video tag.
...
-   ```md
-   [![Open in Coder](https://YOUR_ACCESS_URL/open-in-coder.svg)](https://YOUR_ACCESS_URL/templates/YOUR_TEMPLATE/workspace?disable_params=first_parameter,second_parameter)
-   ```
+### 5. Optional: disable specific parameter fields by including their names as
+
+specified in your template in the `disable_params` search params list
+
+```md
+[![Open in Coder](https://YOUR_ACCESS_URL/open-in-coder.svg)](https://YOUR_ACCESS_URL/templates/YOUR_TEMPLATE/workspace?disable_params=first_parameter,second_parameter)
+```
```
Comment (marcpaq) [inline docs/templates/open-in-coder.md:116]: Delete the section below, `## Devcontainer support`. 

Rationale: It's not (quite) related to Open in Coder and it's properly covered in `docs/templates/devcontainers.md`.

File: docs/templates/creating.md (hunk 1 above)
Comment (marcpaq) [reply docs/templates/creating.md]: Yes. We replaced `concepts.md` with `tour.md`.

**Comment by marcpaq** [comment]
@matifali I've taken a look at the content changes in `docs`. Except for my comments above, it looks good to me.

I've also quickly reviewed the links in `site/src/pages/*/*.tsx`, which seem appropriate (but I'm not an expert on this code).

Nice merge. Thanks!

**Review by bpmct — APPROVED** [review]
LGTM as long as you tested w/ the docs preview

**Comment by matifali** [comment]
links check pass: https://github.com/coder/coder/actions/runs/6548788879/job/17784246419
manually tested docs
